kuma/users/views.py

```diff
@@ -716,7 +720,16 @@ def dispatch(self, request, *args, **kwargs):
         return super(SignupView, self).dispatch(request, *args, **kwargs)
 
 
-signup = redirect_in_maintenance_mode(SignupView.as_view())
+signup_as_view = SignupView.as_view()
+@redirect_in_maintenance_mode
+def signup(request, *args, **kwargs):
+    """Wrapper function just so we can trigger a track_event() call"""
+    if request.session.get('sociallogin_provider'):
+        track_event(
+            CATEGORY_SIGNUP_FLOW,
+            ACTION_PROFILE_AUDIT,
+            request.session['sociallogin_provider'])
+    return signup_as_view(request, *args, **kwargs)
```

Comment:
The problem with this code is that this view is called for both GET and POST (POST when the `Create profile` button is clicked), so we will see multiple events when the user tries to create a profile (at least two events, but more if there are errors on the submitted profile page, for example, forgetting to agree to Mozilla's terms and privacy policy). So I'd recommend replacing the code above with the following changes (which I tested locally):
```python
    def get(self, request, *args, **kwargs):
        if request.session.get('sociallogin_provider'):
            track_event(
                CATEGORY_SIGNUP_FLOW,
                ACTION_PROFILE_AUDIT,
                request.session['sociallogin_provider'])
        return super().get(request, *args, **kwargs)


signup = redirect_in_maintenance_mode(SignupView.as_view())
```
so restoring the `signup` function to its original state, and adding a `get` method to the `SignupView` class so we only trigger the `profile-audit` event on GET.